.ci/checker-framework.groovy

```diff
@@ -473,4 +471,112 @@ class CheckerFrameworkError implements Comparable<CheckerFrameworkError> {
         return toXmlString
     }
 
+    @Override
+    boolean equals(other) {
+        if (this.is(other)) {
+            return true
+        }
+
+        if (getClass() != other.class) {
+            return false
+        }
+
+        CheckerFrameworkError that = (CheckerFrameworkError) other
+
+        boolean isEqualExceptDetails = fileName == that.fileName &&
+                lineContent == that.lineContent &&
+                specifier == that.specifier
+
+        if (that.isUnstable()) {
+            // do not compare details for unstable suppression
+            return isEqualExceptDetails &&
+                    isFuzzyMatch(message, that.message)
+
+        }
+
+        return isEqualExceptDetails &&
+                message == that.message &&
+                details == that.details
+    }
+
+    @Override
+    int hashCode() {
+        int result
+        result = (fileName != null ? fileName.hashCode() : 0)
+        result = 31 * result + (specifier != null ? specifier.hashCode() : 0)
+        result = 31 * result + (message != null ? message.hashCode() : 0)
+        result = 31 * result + (details != null ? details.hashCode() : 0)
```

Comment:
This won't work if `hashCode` is actually required and used.

> If two objects are equal according to equals() method, then their hash code must be same.

Reply:
Good point

Reply:
Removed.

Reply:
As long as you override `hashCode` and don't implement similar logic between it and equals, it will be an issue. Removing details and message from just this method won't "Fix" the underlying issue.

Is this hashCode actually used/needed?

Reply:
No, I just figured codenarc or something else would have a warning about overriding equals but not hashcode.

Reply:
I've removed it, let's see if there are any warnings.

Reply:
https://github.com/checkstyle/checkstyle/blob/master/.ci/checker-framework.groovy#L163
>     final Set<CheckerFrameworkError> errors = new HashSet<>()

It is a hash object. You will need a valid `hashCode` and `equals`.

Reply:
> Another question, do we actually have a reason for it to be `equals` or can it be it's own thing like `fuzzyEquals`?

Set.contains relies on equals.